src/main/java/io/dropwizard/actors/RabbitmqActorBundle.java

```diff
@@ -34,4 +42,21 @@ public void initialize(Bootstrap<?> bootstrap) {
     }
 
     protected abstract RMQConfig getConfig(T t);
+
+    /**
+     * Provides metric registry for instrumenting RMQConnection. If method returns null, metrics collector for
+     * RabbitMQ connection is not initialized.
+     */
+
+    protected MetricRegistry getMetricRegistry(T t) {
+        return null;
+    }
+
+    /**
+     * Provides implementation for {@link ExecutorServiceProvider}. Should be overridden if custom executor service
+     * implementations needs to be used. For e.g. {@link com.codahale.metrics.InstrumentedExecutorService}.
+     */
+    protected ExecutorServiceProvider getExecutorServiceProvider(T t) {
```

Comment:
you _have_ T here .. that means you can getConfig() and create the pool here. That ExecutorServiceProvider class is useless. Remove it.